jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java

```diff
@@ -166,4 +133,44 @@ public void removeSessionListener(WebSocketSessionListener listener)
     {
         return sessionListeners;
     }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        if (sslContextFactory == null)
+        {
+            if (policy.getBehavior() == WebSocketBehavior.CLIENT)
+            {
+                this.sslContextFactory = new SslContextFactory.Client();
+            }
+            else if (policy.getBehavior() == WebSocketBehavior.SERVER)
+            {
+                this.sslContextFactory = new SslContextFactory.Server();
+            }
+        }
+
+        if (executor == null)
+        {
+            QueuedThreadPool threadPool = new QueuedThreadPool();
+            String behavior = "Container";
+            if (policy != null)
```

Comment:
Why this null check here? You would have already gotten a NPE at line 142 above, no?